server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -722,6 +716,38 @@ public Entry(
                 userMetadata,
                 version,
                 null,
+                byRepoShardIdBuilder,
+                res
+            );
+        }
+
+        private static Entry createClone(
```

Comment:
I am not sure what is it cloning? May be simplify to `create`?

Reply:
Cloning is when you make a partial (only a given set of indices) or full copy of another snapshot. The "clone" part is quite fundamental here to differentiate from snapshot creation :)